### PR TITLE
Feature: Tipping support for medium.com

### LINKED ISSUE #663: Add tipping support for Medium profiles and articles
"[Medium](https://medium.com/) is a place to write, read, and connect". Similar to the [tipping feature on Twitter or Vimeo](https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/content-script/batteries/VimeoVideo.ts) Alby should support tipping of articles and profiles on [Medium.com](https://medium.com/). Alby can extract lightning address details from the article and author profile

**Additional information**
Medium also offers the possibility to add custom domains which would need Alby to search for Lightning Addresses, LNURL, etc on every website. As discussed in the community call today we limit support to blogs on medium.com and will tackle custom domains in a separate issue. 

I am going to work on this issue. 🙌

### PR DESCRIPTION
**Pull request by reneaaron**
Fixes #663 

#### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

#### Describe the changes you have made in this PR -

This PR adds tipping support for medium.com. 

#### How Has This Been Tested?

There are 3 places which are currently supported by this battery:

- [x] [Medium post](https://medium.com/@getalby-test/getalby-test-coinos-io-1502ee1cd33f)
- [x] [Author page](https://medium.com/@getalby-test)
- [x] [Author about page](https://medium.com/@getalby-test/about)

Since the short bio is limited to 160 characters only Lightning Addresses are supported for now.

#### Checklist:

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

**Comment by reneaaron** [comment]
@bumi Thanks for your feedback during the call today. I had a closer look at the username structure of URLs on medium.com. There are basically 4 scenarios: 
1. Username with @ (https://medium.com/@getalby-test/getalby-test-coinos-io-1502ee1cd33f)
2. Username without @ (https://medium.com/codica/16-leading-trends-in-ux-ui-design-you-should-know-in-2022-a22648042114)
3. Username on Subdomain (https://techwithdom.medium.com/why-i-moved-away-from-notion-c5ca0303959)
4. Custom domains (https://blog.prototypr.io, not supported)

I'll update the MR so we can support scenario 2 & 3 too.

File: src/extension/content-script/batteries/Medium.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const battery = (): void => {
+  const authorLinkSelector = getAuthorLinkSelector();
+  if (!authorLinkSelector) return;
+
+  const shortBio = document.querySelector<HTMLParagraphElement>(
+    `${authorLinkSelector}~p:last-of-type`
+  )?.innerText;
+
+  if (shortBio) {
+    const match = shortBio.match(/(⚡:?)\s?(\S+@\S+)/i);
```
Comment (bumi) [inline src/extension/content-script/batteries/Medium.ts:15]: I think we should use the same regex as in the others and also allow lnurlp: or lightning: there. 

https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/content-script/batteries/YouTubeVideo.ts#L25

what do you think? 
ideally we have one place where we define that regex

File: src/extension/content-script/batteries/Medium.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    if (match) {
```
Comment (bumi) [inline src/extension/content-script/batteries/Medium.ts:17]: I mostly prefer the guards to exit early, e.g.: 
```
if (!match) {
  return;
}
``

**Review by bumi — APPROVED** [review]

**Comment by reneaaron** [comment]
@bumi Thanks for your feedback! I agree on the pattern and fixed the structuring of the code. Please let me know if there still is anything that needs to be done to get this merged.

**Pull request merged**